index.js

```diff
@@ -32,105 +33,188 @@ function loader (registryOrVersion) {
+              can_place_on: item.canPlaceOn,
+              can_destroy: item.canDestroy
+              // blocking_tick: 0, // TODO
+            }
+          }
+
+          return networkItem
+        }
       }
       throw new Error("Don't know how to serialize for this mc version ")
     }
 
-    static fromNotch (item) {
+    static fromNotch (networkItem, stackId) {
       if (registry.supportFeature('itemSerializationWillOnlyUsePresent')) {
-        if (item.present === false) return null
-        return new Item(item.itemId, item.itemCount, item.nbtData)
+        if (networkItem.present === false) return null
+        return new Item(networkItem.itemId, networkItem.itemCount, networkItem.nbtData)
       } else if (registry.supportFeature('itemSerializationAllowsPresent')) {
-        if (item.itemId === -1 || item.present === false) return null
-        return new Item(item.itemId, item.itemCount, item.nbtData)
+        if (networkItem.itemId === -1 || networkItem.present === false) return null
+        return new Item(networkItem.itemId, networkItem.itemCount, networkItem.nbtData)
       } else if (registry.supportFeature('itemSerializationUsesBlockId')) {
-        if (item.blockId === -1) return null
-        return new Item(item.blockId, item.itemCount, item.itemDamage, item.nbtData)
+        if (networkItem.blockId === -1) return null
+        return new Item(networkItem.blockId, networkItem.itemCount, networkItem.itemDamage, networkItem.nbtData)
+      } else if (registry.type === 'bedrock') {
+        if (registry.version['<']('1.16.220')) {
+          // unsure about this, different packets use slightly different formats, but everything
+          // in the item field stays the same - just sometimes it isn't in the item field
+          const item = new Item(
+            networkItem.network_id,
+            networkItem.auxiliary_value & 0xff,
+            networkItem.auxiliary_value >> 8,
+            networkItem.nbt,
+            stackId
+          )
+          item.blocksCanPlaceOn = networkItem.item?.can_place_on ?? networkItem.can_place_on
+          item.blocksCanDestroy = networkItem.item?.can_destroy ?? networkItem.can_destroy
+          return item
+        } else {
+          const item = new Item(
+            networkItem.network_id,
+            networkItem.count,
+            networkItem.metadata,
+            networkItem.extra.nbt,
+            networkItem.stack_id
+          )
+          item.blocksCanPlaceOn = networkItem.extra.canPlaceOn
+          item.blocksCanDestroy = networkItem.extra.canDestroy
+          return item
+        }
       }
       throw new Error("Don't know how to deserialize for this mc version ")
     }
 
     get customName () {
-      if (Object.keys(this).length === 0) return null
-      return this?.nbt?.value?.display?.value?.Name?.value ?? null
+      if (Object.keys(this).length === 0 || !this.nbt) return null
+
+      return nbt.simplify(this.nbt).display?.Name ?? null
```

Comment:
This get function can be simplified to one line (the other checks are not necessary with optional chanining):
```
return this.nbt?.value?.display?.value?.Name?.value
```